### PR TITLE
tts plugin keep across items TCA-381

### PR DESCRIPTION
**Pull request by lecosson**
**Related to**
https://oat-sa.atlassian.net/browse/TCA-381

**Description**
test taker wants to have the TTS tool defaulted to on when I begin the test and stay on (unless turn it off) across all items of the test.

**How to test**
1. prepare ACT instance
2. link or replace package with this branch
3. navigate to test with some APIP items
4. check APIP plugin state and position across items

**Related PR's**
https://github.com/oat-sa/extension-tao-testqti/pull/1767

### CODE REVIEW

File: src/plugins/tools/apipTextToSpeech/plugin.js [hunk 1]
```diff
@@ -316,6 +318,10 @@ export default pluginFactory({
 
                 getTTSComponent().setMediaContentData(ttsApipData);
                 this.show();
+                if (!this.getState('sleep')) {
+                    this.setState('enabled', true);
```
Comment (jsconan) [inline src/plugins/tools/apipTextToSpeech/plugin.js:322]: Why not use `this.enable()` instead? The state `enabled` is managed internally by the methods `enable()` and `disable()` (cf. [the plugin abstraction](https://github.com/oat-sa/tao-core-sdk-fe/blob/8cf13ba0c121d845d44c6e675084920cabacf6ef/src/core/plugin.js#L325:L344)).

**Comment by ampaveliev** [comment]
Looks ok, but I can't test it locally. Because I don't understand how should I prepare my local instance and what kind of items are called APIP:

> 1. prepare ACT instance
> 2. ...
>  3. navigate to test with some APIP items

File: src/plugins/tools/apipTextToSpeech/plugin.js [hunk 2]
```diff
@@ -224,6 +225,7 @@ export default pluginFactory({
         // Handle plugin button click
         this.button.on('click', (e) => {
             e.preventDefault();
+            this.setState('sleep', false);
```
Comment (jsconan) [inline src/plugins/tools/apipTextToSpeech/plugin.js:228]: Only when clicking on the button? Maybe could it be better to have this when doing the action instead or am I mistaken? (when catching the event)

**Comment by lecosson** [comment]
> Looks ok, but I can't test it locally. Because I don't understand how should I prepare my local instance and what kind of items are called APIP:
> 
> > 1. prepare ACT instance
> > 2. ...
> > 3. navigate to test with some APIP items

text example added to jira

File: src/plugins/tools/apipTextToSpeech/plugin.js (hunk 1 above)
Comment (lecosson) [reply src/plugins/tools/apipTextToSpeech/plugin.js:322]: there is `enable` method relates to button, not to plugin

File: src/plugins/tools/apipTextToSpeech/plugin.js (hunk 2 above)
Comment (lecosson) [reply src/plugins/tools/apipTextToSpeech/plugin.js:228]: yes, has a reason, fixed

**Comment by lecosson** [comment]
> Code review only.
> Could you add the package-lock file ?
> 
> Unfortunately the unit tests are failing (timeout on CLI, and error message in browser)
> image

fixed

File: src/plugins/tools/apipTextToSpeech/plugin.js [hunk 3]
```diff
@@ -316,6 +318,10 @@ export default pluginFactory({
 
                 getTTSComponent().setMediaContentData(ttsApipData);
                 this.show();
+                if (!this.getState('sleep')) {
+                    this.setState('enabled', true);
+                    toggleTool();
```
Comment (jsconan) [inline src/plugins/tools/apipTextToSpeech/plugin.js:323]: This function is calling `enablePlugin()`, which is recreating the keyNavigation. This leads to inconsistency and make the unit tests failing.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@oat-sa/tao-test-runner-qti",
-    "version": "2.1.1",
+    "version": "2.1.3",
```
[the remark this answers is not in the document]
Comment (lecosson) [reply package.json:3]: 2.1.2 already released

File: src/plugins/tools/apipTextToSpeech/plugin.js (hunk 3 above)
Comment (lecosson) [reply src/plugins/tools/apipTextToSpeech/plugin.js:323]: line added to test, test is passed

**Pull request merged**